framework/src/org/checkerframework/framework/util/MultiGraphQualifierHierarchy.java

```diff
@@ -664,30 +664,37 @@ assert getTopAnnotation(a1) == getTopAnnotation(a2) :
                 "], a2: " + a2 + " [top: " + getTopAnnotation(a2) + "]";
 
         Set<AnnotationMirror> outset = AnnotationUtils.createAnnotationSet();
-        for (AnnotationMirror a1Super : findSmallestTypes(supertypesMap.get(a1))) {
+        for (AnnotationMirror a1Super : supertypesGraph.get(a1)) {
```

Comment:
Depending on the naming of the qualifiers, the LUB of two non-polymorphic qualifiers was, at times, calculated to be a polymorphic qualifier (even though that shouldn't happen). The super types graphs and maps all use alphabetic ordering to sort the set of super types according to the package and class name of each qualifier. The outset also maintains alphabetic ordering. In the old logic, if there's more than 1 qualifier in the outset, then it selects the first in the set to be the LUB. In the new logic, it select the first non-polymorphic qualifier to be the LUB.

I was able to trigger this bug through externally loaded units qualifiers (using -Aunits or -AunitsDirs), as their fully qualified class names began with "qual.__" which caused them to be sorted later than "org. ... .PolyUnit". The code changes here fixes this bug.

It can be made into a separate pull request if desired.